### ISSUE TITLE
Local functions: Some selections around local functions fail extract method refactoring.

### ISSUE THREAD

**Issue by balajikris** [dotnet/roslyn#8719]
When local functions are involved while extracting new method, I tend to run into situations where the refactoring fails with an abstract message - "one or more errors occurred".

Below are a couple of selections that reproduce the issue. I've seen this with many other selections as well, but haven't narrowed down to exactly what causes it yet.

My guess is this is a last check after analysis phase where the refactoring ran into unexpected errors and bails with an abstract message.

``` C#
void Outer()
{
        // begin select 1
        void LocalFunc()
        {
            Console.Write(2);
        }
        /* select 2 */ new Action(LocalFunc)(); /* select 2*/
        // end select 1
 }
```

![image](https://cloud.githubusercontent.com/assets/4213867/13061004/13598662-d3ea-11e5-885b-367c4209d923.png)

**Comment by DustinCampbell** [comment]
Related to #8718 and #8717

**Comment by agocke** [comment]
Fixed by #16021
